php/src/Contracts/Signer.php

```diff
@@ -0,0 +1,16 @@
+<?php
+declare(strict_types=1);
+
+namespace TrueLayer\Signing\Contracts;
+
+use Jose\Component\Core\JWK;
+
+interface Signer extends Jws
+{
+    public static function signWithKey(string $kid, JWK $jwk): self;
+    public static function signWithPem(string $kid, string $pem, ?string $passphrase): self;
+    public static function signWithPemBase64(string $kid, string $pemBase64, ?string $passphrase): self;
+    public static function signWithPemFile(string $kid, string $path, ?string $passphrase): self;
```

Comment:
May be worth making the passphrase default to null so you don't have to set the last param to null yourself
```suggestion
    public static function signWithPem(string $kid, string $pem, ?string $passphrase = null): self;
    public static function signWithPemBase64(string $kid, string $pemBase64, ?string $passphrase = null): self;
    public static function signWithPemFile(string $kid, string $path, ?string $passphrase = null): self;
```

Reply:
Did we decide not make this change?

Reply:
I think we've done it?

Reply:
No, the params are nullable but they don't default to null currently. So you need to pass in null for the third arg yourself.

>PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function TrueLayer\Signing\Signer::signWithPemFile(), 2 passed in test.php on line 9 and exactly 3 expected in TrueLayer/truelayer-signing/php/src/Signer.php:73